### PR TITLE
Add `table.len` method.

### PR DESCRIPTION
**Pull request by Andrey2470T**
If you want to retrieve a length of a list with using `#` operator, it works fine. However, if your table contains still fields with non-integer keys, they are ignored by those operator. It obviously will give an incorrect result. My small method `table.len()` counts **all** fields independing on their keys types.  

## To do
This PR is Ready for Review.

## How to test
Copypaste the function code from the source here and run this code:
```
local random_t = {mesh = "mesh.b3d", texture = "texture.png", 43, 56.5, function() return "nothing" end, {1, 2, 3}, nil}

print(table.len(random_t))
```

### CODE REVIEW

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -486,6 +486,17 @@ function table.copy(t, seen)
 end
 
 
+function table.len(t)
+    local counter = 0
```
Comment (Andrey2470T) [inline builtin/common/misc_helpers.lua:490]: I don't understand why when I upload my code on github, all tabs are displayed incorrectly. In my original code the tabs look like aligned with other parts of the file.

Comment (Desour) [reply builtin/common/misc_helpers.lua:490]: https://dev.minetest.net/Lua_code_style_guidelines
Indent with tabs.

Comment (Andrey2470T) [reply builtin/common/misc_helpers.lua:490]: I suppose two tabs?

**Comment by paramat** [comment]
Discussion in IRC http://irc.minetest.net/minetest-dev/2020-10-07#i_5739976
Seems low priority.
If table length for non-numerical keys is needed, it would be more efficient to increment a counter in the mod's Lua code instead of looping through the table as this function does.
I am not sure this is worth adding.

**Comment by Andrey2470T** [comment]
> Discussion in IRC http://irc.minetest.net/minetest-dev/2020-10-07#i_5739976
> Seems low priority.
> If table length for non-numerical keys is needed, it would be more efficient to increment a counter in the mod's Lua code instead of looping through the table every time you need the table length, as this function does.
> I do not think this is worth adding.

Wait, but the counter can not be incremented without looping over a table, because the real table length is unknown if it concerns to the associative arrays. There are no other ways.

**Comment by paramat** [comment]
Yes, incrementing a counter cannot always be used, only sometimes.

However, a function needs to be frequently used and useful to many people to be included in builtin, the IRC discussion explains why this proposed function probably does not qualify.

**Pull request closed without merging**